Assets/Countly/Plugins/CountlySDK/Services/UserDetailsCountlyService.cs

```diff
@@ -332,23 +444,25 @@ public void PushUnique(string key, string[] value)
         /// </summary>
         /// <param name="key">String with property name for array property</param>
         /// <param name="value">array with values to remove from array</param>
-        public void Pull(string key, string[] value)
+        private void PullInternal(string key, string[] value)
         {
-            if (string.IsNullOrEmpty(key)) {
-                Log.Warning("[UserDetailsCountlyService] Pull : key '" + key + "'isn't valid.");
+            if (!_consentService.CheckConsentInternal(Consents.Users)) {
```

Comment:
you could maybe extract those validations to the a function and use all of the internal methods. because they are validation same things. By that way, we could minimize duplication